### PR TITLE
Emit metrics on reschedule later decisions as nomad.client.allocs.reschedule

### PR DESCRIPTION
**Pull request by joel0**
We implemented this as a companion to #10236 for metrics visibility into rescheduler decisions and limits.

I do not know whether this meets the quality standard to be upstreamed.  If you have ideas on how to make this upstream worthy, I'd be willing to make changes.  I was unable to find a metrics loop that was more appropriate than this with the access to the required alloc info.

### CODE REVIEW

**Comment by tgross** [comment]
Thanks @joel0! We'll take a look -- this is definitely the _sort_ of thing we've been wanting to do to improve scheduler observability.

**Comment by benbuzbee** [comment]
@tgross just wondering if you folks have had a chance to think about these more. We find them very useful and would be happy to polish the impl to your specifications if it helps

File: scheduler/reconcile.go [hunk 1]
```diff
@@ -953,9 +954,30 @@ func (a *allocReconciler) handleDelayedLost(rescheduleLater []*delayedReschedule
...
 	a.result.desiredFollowupEvals[tgName] = evals
 
 	return allocIDToFollowupEvalID
 }
+
+// emitRescheduleInfo emits metrics about the reschedule decision of an evaluation. If a followup evaluation is
+// provided, the waitUntil time is emitted.
+func emitRescheduleInfo(alloc *structs.Allocation, followupEval *structs.Evaluation) {
+	// Emit short-lived metrics data point. Note, these expire and stop emitting after about a minute.
+	baseMetric := []string{"client", "allocs", "reschedule"}
```
Comment (tgross) [inline scheduler/reconcile.go:969]: This name would make these fall under [Client Allocation Metrics](https://www.nomadproject.io/docs/operations/metrics-reference#allocation-metrics) that we emit on the clients only. We have few metrics in the scheduler outside the RPC handlers (something we want to fix), so I don't have a good example of a scheduler-internal one. Maybe `[]string{"scheduler", "allocs", "rescheduled"}`?

**Review by tgross — CHANGES_REQUESTED** [review]
Sorry for the delay @benbuzbee @joel0; we've been poking around the scheduler a bunch lately and particular around terminology so I wanted to make sure we're not emitting metrics that are misleading to operators.

I've left a comment about the metric name. But otherwise the implementation for this LGTM. We'll want to add a line to the [metrics reference](https://www.nomadproject.io/docs/operations/metrics-reference) docs and a changelog entry, but I can follow up with those if y'all would like.

**Comment by tgross** [comment]
Actually, I'm going to go ahead and merge this as-is and do the changelog and docs myself... there's a gap in the definitions on that page I want to fill in.

**Review by tgross — APPROVED** [review]
Actually, I'm going to go ahead and merge this as-is and do the changelog and docs myself... there's a gap in the definitions on that page I want to fill in.

LGTM

File: scheduler/reconcile.go (hunk 1 above)
Comment (tgross) [reply scheduler/reconcile.go:969]: fixed the name in https://github.com/hashicorp/nomad/pull/11790

**Pull request merged**